lib/seadsa/DsaLibFuncInfo.cc

```diff
@@ -33,20 +41,81 @@ void DsaLibFuncInfo::getAnalysisUsage(llvm::AnalysisUsage &AU) const {
+  auto specFnLink = m_specLang->getFunction("sea_dsa_link");
+  auto specFnAccess = m_specLang->getFunction("sea_dsa_access");
+
+  assert(specFnModify != nullptr && "Could not find sea_dsa_set_modified");
+  assert(specFnRead != nullptr && "Could not find sea_dsa_set_read");
+  assert(specFnCollapse != nullptr && "Could not find sea_dsa_set_heap");
+  assert(specFnHeap != nullptr && "Could not find sea_dsa_collapse");
+  assert(specFnP2I != nullptr && "Could not find sea_dsa_set_ptrtoint");
+  assert(specFnI2P != nullptr && "Could not find sea_dsa_set_inttoptr");
+  assert(specFnExtern != nullptr && "Could not find sea_dsa_set_external");
+  assert(specFnAlias != nullptr && "Could not find sea_dsa_alias");
+  assert(specFnMk != nullptr && "Could not find sea_dsa_mk");
+  assert(specFnLink != nullptr && "Could not find sea_dsa_link");
+  assert(specFnAccess != nullptr && "Could not find sea_dsa_access");
+
+  m_specModule->getOrInsertFunction(specFnModify->getName(),
+                                    specFnModify->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnRead->getName(),
+                                    specFnRead->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnCollapse->getName(),
+                                    specFnCollapse->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnHeap->getName(),
+                                    specFnHeap->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnP2I->getName(),
+                                    specFnP2I->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnI2P->getName(),
+                                    specFnI2P->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnExtern->getName(),
+                                    specFnExtern->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnAlias->getName(),
+                                    specFnAlias->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnMk->getName(),
+                                    specFnMk->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnLink->getName(),
+                                    specFnLink->getFunctionType());
+  m_specModule->getOrInsertFunction(specFnAccess->getName(),
+                                    specFnAccess->getFunctionType());
+}
+
 void DsaLibFuncInfo::initialize(const llvm::Module &M) const {
 
   using namespace llvm::sys::fs;
   using namespace llvm;
 
-  if (m_isInitialized || (XSpecFiles.empty() && !XUseClibSpec)) return;
+  if (m_isInitialized ||
+      (XSpecFiles.empty() && !XUseClibSpec && XGenSpecs.empty()))
+    return;
   m_isInitialized = true;
-  auto dl = M.getDataLayout();
 
   SMDiagnostic err;
+  auto &dl = M.getDataLayout();
+  auto &ctx = M.getContext();
 
   auto exePath = getMainExecutable(nullptr, nullptr);
   StringRef exeDir = llvm::sys::path::parent_path(exePath);
 
+  const StringRef specLangRel = "../lib/sea_dsa.ll";
```

Comment:
Find better location for spec files. Now they would go to /usr/lib if seahorn is installed globally. Should besomewhere under /usr/shared